src/views/ApartmentApp/components/StudentApplication/index.js

```diff
@@ -14,171 +14,179 @@ import AlertDialogBox from '../../../../components/AlertDialogBox';
+  </Card>
+);
+
+const StudentApplication = ({ userProfile }) => {
+  const [loading, setLoading] = useState(true);
+  const [saving, setSaving] = useState(false);
+
+  const [applicationID, setApplicationID] = useState(-1); // Default value of -1 indicate to backend that the application ID number is not yet known
+  const [editorUsername, setEditorUsername] = useState(null); // The username of the application editor
+  const [applicants, setApplicants] = useState([]);
+  const [preferredHalls, setPreferredHalls] = useState([{ HallRank: 1, HallName: '' }]);
+  // const [offCampusProgramInfo, setOffCampusProgramInfo] = useState(new Map());
+
+  const [applicationCardsOpen, setApplicationCardsOpen] = useState(false);
+  const [newEditorProfile, setNewEditorProfile] = useState(null); // Stores the StudentProfileInfo of the new editor before the user confirms the change
+  const [changeEditorDialogOpen, setChangeEditorDialogOpen] = useState(false);
+  const [submitDialogOpen, setSubmitDialogOpen] = useState(false); // Use this for submitting app (later feature)
+  const [snackbarOpen, setSnackbarOpen] = useState(false);
+  const [snackbarText, setSnackbarText] = useState('');
+  const [snackbarSeverity, setSnackbarSeverity] = useState('');
+  const [saveButtonAlertTimeout, setSaveButtonAlertTimeout] = useState(null);
+
+  // const peopleSearchRef = useRef();
+
+  let offCampusProgramInfo = new Map();
+  applicants.forEach((profile) => offCampusProgramInfo.set(profile.AD_Username, ''));
 
   /**
-   * Loads the user's profile info only once (at start)
+   * Attempt to load an existing application from the database if one exists
    */
-  async loadProfile() {
-    this.setState({ loading: true });
-    try {
-      let applicants = this.state.applicants;
-      applicants.push(this.props.userProfile);
-      this.setState({ applicants });
-      this.setState({ loading: false });
-    } catch (error) {
-      // Do Nothing
-    }
-  }
+  useEffect(() => {
+    loadSavedApplication();
+
+    // eslint-disable-next-line
+  }, []);
 
-  async loadSavedApplication() {
+  const loadSavedApplication = async () => {
     // TODO: Implement this once save/load of application data has been implemented in the backend
-    if (!this.state.editorUsername) {
-      this.setState({ editorUsername: this.props.userProfile.AD_Username });
+    setLoading(true);
+    // Check if the current user is on an application. Returns the application ID number if found
+    let newApplicationID = -1; // await housing.getApplicationID();
+    if (newApplicationID !== null && newApplicationID !== -1) {
+      setApplicationID(newApplicationID);
+      let applicationDetails = await housing.getApartmentApplication(newApplicationID);
+      if (applicationDetails) {
+        if (applicationDetails.Username) {
+          setEditorUsername(applicationDetails.Username);
+        }
+        if (applicationDetails.Applicants) {
+          setApplicants(applicationDetails.Username);
```

Comment:
Is this supposed to be Username? My instinct would be that you would set applicants to applicationDetails.Applicants.